### ISSUE TITLE
Is MA0127 intentionally broad or is it a bug?

### ISSUE THREAD

**Issue by sarahelsaig** [meziantou/Meziantou.Analyzer#1159]
### Version of the Meziantou.Analyzer NuGet package

3.0.84

### Rule Identifier

MA0127

### Target Framework

net8.9

### C# Language version

C# 12

### Description

According to the rule description
```csharp
variable is "constant"
```
is bad. I understand that it's a misuse of pattern matching for string comparison. But what about more complex patterns that contain string value matching? Currently these also trigger the rule and we were wondering if this is an oversight. If it's intentional, it would be good to mention in the docs page for MA0127 that string 'is` should be avoided in all patterns and not just in the example where it's the whole pattern.

### Reproduction Steps

These deeper patterns also trigger the rule

```c#
variable is { Metadata.Type: "MyShape" }
variable is true or "true"
```

even though now the patterns do much more than just a string comparison.

**Comment by meziantou** [comment]
The rule is hidden by default, so it's only useful to provide a codefix. The main idea is to help converting a simple pattern matching on string values to an ignore case comparison:

```c#
_ = variable is "value"; // Ordinal
_ = string.Equals(variable, "value", StringComparison.OrdinalIgnoreCase);
```

You can update the default level if you want to be more explicit when comparing strings.

It should not report on non-string values.

**Comment by sarahelsaig** [comment]
The rule shows up for me as a warning even though I have not explicitly changed its severity. After some tinkering, I was able to find a minimal way to reproduce the phenomenon.

<img width="1759" height="990" alt="Image" src="https://github.com/user-attachments/assets/15263653-e3ae-476c-9247-c57a8441dc69" />

Is the rule meant to enable when `dotnet_analyzer_diagnostic.category-Usage.severity = warning` is present?

**Comment by meziantou** [comment]
The rule has the category `Usage`, so maybe that the reason why it's a warning on your side. Note that the configuration part is handled by Roslyn, not the code in this project.

**Comment by sarahelsaig** [comment]
I see, then I will just explicitly reset the rule to its default level in config. Thanks for bearing with me.
